### PR TITLE
[Author] Update PreloadJS package.json w/ npm auto-update config

### PR DESCRIPTION
**Pull request by wdamien**
Added npmFileMap for auto update support.

Pull request for issue: #
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/CreateJS/PreloadJS
 * Official website (optional, not the repository): http://createjs.com/preloadjs
 * NPM package url (optional): https://www.npmjs.com/package/preloadjs
 * License and its reference: MIT - https://github.com/CreateJS/PreloadJS/blob/master/LICENSE.txt
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 141
   - Count of stars: 2, 067
   - Count of forks: 514
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ ] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [ ] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [ ] The parent of the commit(s) in the PR is not old than 3 days.
 * [ ] Pull request is sending from a non-master branch with meaningful name.
 * [ ] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sufuf3** [comment]
Please take a look at #11851, this package.json' order as follows:
(This is helpful to us that if we use `git diff` to review this PR. Also, we have to keep version field. Thanks.)
Similar to #11851
Could you modify this? thanks.

```js

{
  "name": "PreloadJS",
  "filename": "preloadjs.min.js",
  "version": "0.6.0",
  "description": "PreloadJS makes it easy to preload your assets: images, sounds, JavaScript, fonts, JSON, and text data.",
  "homepage": "https://createjs.com/preloadjs/",
  "keywords": [
    "createjs",
    "preloadjs",
    "gskinner",
    "preload",
    "preloading",
    "progress",
    "XHR",
    "javascript",
    "html5"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/CreateJS/PreloadJS.git"
  },
  "license": "MIT",
  "author": "gskinner.com, inc.",
  "npmName": "preloadjs",
  "npmFileMap": [
    {
      "basePath": "lib",
      "files": [
        "**/*"
      ]
    }
  ]
}
```

**Comment by sufuf3** [comment]
Helped rebase master and squash the commit.

@wdamien May I ask that which auto-update do you like to use, npm or git?
Because I found that there are 3 releases versions from npm, and 11 releases versions from GitHub.
As usual, we prefer to use more releases versions as the auto-update option.
But if you think npm is stable than GitHub repo, then npm is a good choice.

Thank you.

**Comment by wdamien** [comment]
@sufuf3 The old versions don't match your required folder structure  (and we won't be changing the old tags). So going forward NPM is fine, since we will keep both GitHub and NPM in sync.

File: ajax/libs/PreloadJS/package.json [hunk 1]
```diff
@@ -2,22 +2,32 @@
...
-    "lannymcnie",
-    "wdamien"
+  "author": "gskinner.com, inc.",
+  "homepage": "https://createjs.com/preloadjs/",
+  "npmName": "preloadjs",
+  "npmFileMap": [
+    {
+      "basePath": "lib",
+      "files": [
+        "**/*"
```
Comment (extend1994) [inline ajax/libs/PreloadJS/package.json]: Need to modify this to `**/!(README.md)` ( or `**/!(*.md)` if you could put any other markdown file under `dist` in the future) to exclude `README.md`

Comment (sufuf3) [reply ajax/libs/PreloadJS/package.json]: @wdamien Please help modify the pattern to `**/!(*.md)`
Thank you.
(Except for this part, I think this PR is GOOD!)

**Comment by PeterDaveHello** [comment]
@extend1994 please help open a new issue about the missing versions, we should focus on package.json info update and auto-update config.

**Comment by extend1994** [comment]
Opened in #12228.

**Pull request merged**